aten/src/ATen/native/mkldnn/xpu/Blas.cpp

```diff
@@ -228,7 +222,12 @@ Tensor& baddbmm_out(
           1.f, alpha.to<float>(), 0.f, attr.kind_with_linear);
     }
   } else {
-    binary = input.dim() < 3 ? input.unsqueeze(0) : input;
+    // We use post_binary here for adding self matrix.
+    // To avoid wrong write, here we clone input for inplace case.
+    if (is_inplace)
+      binary = input.dim() < 3 ? input.unsqueeze(0).clone() : input.clone();
+    else
+      binary = input.dim() < 3 ? input.unsqueeze(0) : input;
     binary = binary.dim() < 3 ? binary.unsqueeze_(0) : binary;
```

Comment:
Do you mean the number of dimensions of `input` could be `1`? So that the ndim of binary will be 3 by invoking twice times of `unsqueeze`.

Reply:
If so, pls. add detailed comments.

Reply:
Yes, this is possible since matmul operators usually need handle broadcast issues.